docs/08_jenkins_testing_enviroment.md

```diff
@@ -0,0 +1,126 @@
+
+[Jenkins](https://jenkins.io/)  
+[Google Cloud Jenkins application](https://github.com/GoogleCloudPlatform/click-to-deploy/blob/master/k8s/jenkins/README.md)
+[Amazon AWS](https://aws.amazon.com/)
+
+# Motivation
+
+> What problem is this project solving?
+
+Creating the Jenkins test environment will provide the data for the Metrics application.
+The Jenkins test environment will give us the ability to test the whole process of importing and displaying the project metrics data.
+
+# Goals
+
+> Identify success metrics and measurable goals.
+
+* Jenkins server is running and available for testing by anyone in the team.
+* Configuration of the Jenkins to run the tests from the test project. 
+
+# Non-Goals
+
+> Identify what's not in scope.
+
+The creation of the test project is not in the scope of this project.
+
+# Design
+
+> Explain and diagram the technical design
+
+`Amazon AWS instance` -> `Jenkins Server`
+
+> Identify risks and edge cases
+
+Running the builds in the separate docker containers could require an additional configuration.
+
+# API
+
+> What will the proposed API look like?
+
+The Jenkins API is described in this article: [Jenkins remote access api](https://wiki.jenkins.io/display/JENKINS/Remote+access+API).
+
+# Dependencies
+
+> What is the project blocked on?
+
+No blockers.
+
+> What will be impacted by the project?
+
+Testing of the Jenkins integration will be impacted.
+
+# Testing
+
+> How will the project be tested?
+
+This project will be tested manually.
+
+# Alternatives Considered
+
+> Summarize alternative designs (pros & cons)
+
+* Setups Jenkins server on Amazon AWS.
+    - Pros:
+         - The AWS instance is free for one year.
+         - Has an excellent [setup guide](https://d1.awsstatic.com/Projects/P5505030/aws-project_Jenkins-build-server.pdf) that explains how to create an instance and deploy the Jenkins server.
+         - Easy to maintain. You could just connect to the AWS instance and use it and use it the way that you would use a computer in front of you.
+    - Cons:
+         - You need to configure the AWS instance and firewall rules on your own, which takes a bit more time in comparison to using the Google Kubernetes deployment.
+         
+         
+* Setup Jenkins on Google Kubernetes Engine.
+    - Pros:  
+         - Easy to set up. The configuration process is fully automated, and it is needed just to configure Jenkins projects.
+           It is needed just to go through the [Jenkins Google cloud setup flow](https://console.cloud.google.com/marketplace/details/google/jenkins).
+         - Jenkins has a [free application](https://console.cloud.google.com/marketplace/details/google/jenkins) on the Google Cloud.
+    - Cons: 
+         - The Kubernetes Engine is the paid feature of the Google Cloud that uses a couple of other paid features like Compute Engine.
+         - Hard to maintain because you cannot just connect to the instance with the Jenkins server to make any changes.
```

Comment:
```suggestion
         - Hard to maintain because you cannot just SSH connect to the instance with the Jenkins server to make any changes. For connection, you need to use CLI "kubectl".
```